protocol/java/org/openyolo/protocol/Hint.java

```diff
@@ -0,0 +1,398 @@
+            return this;
+        }
+
+        /**
+         * Specifies the authentication method for the hint. Must not be null.
+         */
+        public Builder setAuthMethod(@NonNull AuthenticationMethod authMethod) {
+            require(authMethod, notNullValue());
+            mAuthMethod = authMethod;
+            return this;
+        }
+
+        /**
+         * Specifies the display name for the hint. Empty values are treated as null.
+         */
+        public Builder setDisplayName(@Nullable String displayName) {
+            mDisplayName = nullifyEmptyString(displayName);
+            return this;
+        }
+
+        /**
+         * Specifies the display picture URI for the hint, in string form. Empty values are
+         * treated as null. Non-empty values must be valid http(s) URIs.
+         */
+        public Builder setDisplayPictureUri(@Nullable String displayPictureUri) {
+            setDisplayPictureUri(nullifyEmptyString(displayPictureUri) != null
+                            ? Uri.parse(displayPictureUri)
+                            : null);
+            return this;
+        }
+
+        /**
+         * Specifies the display picture URI for the hint, in string form. Non-null values
+         * must be valid http(s) URIs.
+         */
+        public Builder setDisplayPictureUri(@Nullable Uri displayPictureUri) {
+            require(displayPictureUri, nullOr(isWebUri()));
+            mDisplayPictureUri = displayPictureUri;
+            return this;
+        }
+
+        /**
+         * Specifies the generated password for the hint. Empty values are treated as null.
+         */
+        public Builder setGeneratedPassword(@Nullable String generatedPassword) {
+            mGeneratedPassword = nullifyEmptyString(generatedPassword);
+            return this;
+        }
+
+        /**
+         * Specifies the "proof of access" ID token for the hint. Empty values are treated as null.
+         */
+        public Builder setIdToken(@Nullable String idToken) {
+            mIdToken = nullifyEmptyString(idToken);
+            return this;
+        }
+
+        /**
+         * Specifies any additional, non-standard properties associated with the hint.
+         */
+        public Builder setAdditionalProperties(@Nullable Map<String, byte[]> additionalProps) {
+            mAdditionalProps =
+                    AdditionalPropertiesUtil.validateAdditionalProperties(additionalProps);
+            return this;
+        }
+
+        private Builder setAdditionalPropertiesFromProto(
+                @Nullable Map<String, ByteString> additionalProps) {
+            mAdditionalProps =
+                    AdditionalPropertiesUtil.validateAdditionalPropertiesFromProto(additionalProps);
+            return this;
+        }
+
+        /**
+         * Creates the {@link Hint hint} instance with the specified properties.
+         */
+        public Hint build() {
+            return new Hint(
```

Comment:
Done. We should be consistent about this so I'll open an issue to apply this pattern elsewhere.